stdlib/REPL/test/replcompletions.jl

```diff
@@ -2044,3 +2044,8 @@ end
     # If this last test starts failing, that's okay, just pick a new example symbol:
     @test !Base.isexported(Base, :ispublic)
 end
+
+let t = REPLCompletions.repl_eval_ex(:(`a b`), @__MODULE__)
+    @test t isa Core.Const
+    @test t.val == `a b`
+end
```

Comment:
Does this need to be constant folded? I submitted #51581 in order to fix #51548, where the changes now don't allow `Base.cmd_gen` to be constant-folded.

Reply:
It would be nice for https://github.com/JuliaLang/julia/pull/51544, but it's also not a total showstopper. This particular case should still be fine though since the `cmd_gen` call happens at the top-level here, right?

Reply:
Right, if we can concrete-eval `Base.cmd_gen` (, which will be enabled by this PR), it should be fine after all. It can be problematic if concrete-evaluation happens for the callee of `Base.cmd_gen`.

Reply:
I'd be fine with this only working in toplevel expressions, so feel free to go ahead on that front!